bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,322 @@
+                if is_present:
+                    nkeys_present += 1
+                    db_id = int(db_id_str)
+                    # Some don't have gloss_id's
+                    try:
+                        gloss_id = int(gloss_id_str)
+                    except ValueError:
+                        gloss_id = None
+                    is_public = is_public_str.strip().lower() == "true"
+                else:
+                    nkeys_absent += 1
+                    db_id = None
+                    gloss_id = None
+                    is_public = None
+
+                this_all_keys_dict[video_key] = [is_present, db_id, gloss_id, is_public]
+
+        print(f"PRESENT: {nkeys_present} keys", file=sys.stderr)
+        print(f"ABSENT:  {nkeys_absent} keys", file=sys.stderr)
+
+        return this_all_keys_dict
+
+    except FileNotFoundError:
+        print(f"File not found: {cache_file}", file=sys.stderr)
+        exit()
+
+
+# Get all keys from AWS S3
+def get_s3_bucket_raw_keys_list(
+    s3_bucket=AWS_S3_BUCKET, keys_file=S3_BUCKET_RAW_KEYS_FILE
+):
+    print(f"Getting raw AWS S3 keys recursively ({s3_bucket}) ...", file=sys.stderr)
+    with open(keys_file, "w") as f_obj:
+        subprocess.run(
+            [AWSCLIENT, "s3", "ls", f"s3://{s3_bucket}", "--recursive"],
+            env=NEW_ENV,
+            shell=False,
+            check=True,
+            text=True,
+            stdout=f_obj,
+        )
+
+    # Separate out just the key (also strip newline) from date, time, size, key
+    # Put the keys in an in-memory list
+    with open(keys_file, "r") as f_obj:
+        this_s3_bucket_raw_keys_list = [line.split()[3] for line in f_obj]
+    print(
+        f"{len(this_s3_bucket_raw_keys_list)} rows retrieved: {keys_file}",
+        file=sys.stderr,
+    )
+
+    # Write the keys back to the file, for cleanliness
+    with open(keys_file, "w") as f_obj:
+        for line in this_s3_bucket_raw_keys_list:
+            f_obj.write(f"{line}\n")
+
+    return this_s3_bucket_raw_keys_list
+
+
+# Get the video files info from NZSL Signbank
+def get_nzsl_raw_keys_dict(keys_file=NZSL_POSTGRES_RAW_KEYS_FILE):
+    this_nzsl_raw_keys_dict = {}
+    print(
+        f"Getting raw list of video file info from NZSL Signbank ...",
+        file=sys.stderr,
+    )
+    with open(keys_file, "w") as f_obj:
+        subprocess.run(
+            [
+                PGCLIENT,
+                "-t",
+                "-c",
+                "select id as db_id, gloss_id, is_public, videofile from video_glossvideo",
+                f"{DATABASE_URL}",
+            ],
+            env=NEW_ENV,
+            shell=False,
+            check=True,
+            text=True,
+            stdout=f_obj,
+        )
+    with open(keys_file, "r") as f_obj:
+        nzsl_raw_keys_list = f_obj.readlines()
+    print(
+        f"{len(nzsl_raw_keys_list)} rows retrieved: {keys_file}",
+        file=sys.stderr,
+    )
```

Comment:
issue: since we immediately read the file into memory for parsing, I think we should just skip writing this to a file

I personally think it'll be fine to just read the whole stdout into memory, but you should be able to use `subprocess.Popen` with `stdout=subprocess.PIPE` to get an iterator that will let you then read each line one-at-time to be more memory efficient

(I would just start though by reading the entire output into memory)

Reply:
I tried using file-like objects originaly and it just did not work.
However, the intermediate files are probably uneccessary now, except for the caching one which speeds up debugging a lot.
I'll see if I can get rid of the intermediates, shouldn't be difficult.

Reply:
Yeah, no, sorry, that opens a whole kettle of fish to do with subprocess.run.